### PR TITLE
Pivot tables sorting

### PR DESCRIPTION
**Pull request by paulrosenzweig**
The PR wires up @ariya's sorting UI to a new pivot table column setting. 

sorting

@mazameli, I didn't add the sorting in the column header action menu. The code was a bit awkward to refactor into handling both types of sorting (normal and pivot). If you feel strongly, I can definitely add that though.

### CODE REVIEW

**Comment by mazameli** [comment]
This is great! I do think we'll definitely want to add sorting from the column header action menu (at least for row grouping headings, like `User.Source` in your gif). But I think now that we have the ability to sort _at all_, I'd feel okay if we felt the need to wait to do that in 0.38.1, though I'd prefer to not have to wait until 0.39 for it.

(Side note: since we don't have a header that displays the name of a pivoted column, adding an intuitive way to sort pivoted columns outside of the settings sidebar might be tricky, and possibly not even worth doing unless you have a great idea.)

File: frontend/src/metabase/lib/data_grid.js [hunk 1]
```diff
@@ -375,6 +383,23 @@ function updateValueObject(row, indexes, seenValues, collapsedSubtotals = []) {
     if (seenValue === undefined) {
       seenValue = { value, children: [], isCollapsed };
       currentLevelSeenValues.push(seenValue);
+      const sortOrder = getIn(columnSettings, [index, COLUMN_SORT_ORDER]);
+      if (sortOrder) {
+        currentLevelSeenValues.sort((a, b) => {
```
Comment (ariya) [inline frontend/src/metabase/lib/data_grid.js:388]: This block is a bit tricky to follow along. A potential follow-up will be a more self-contained version of the block (a smaller pure function?), along with the corresponding unit tests.

**Review by ariya — APPROVED** [review]
Looks clean and solid :+1:

File: frontend/src/metabase/lib/data_grid.js (hunk 1 above)
Comment (paulrosenzweig) [reply frontend/src/metabase/lib/data_grid.js:388]: Yeah good call. I'll extract that now. The branches (strings with accents, numbers) are covered through the unit tests of `multiLevelPivot`, but pulling out a distinct function would make the code more clear.

**Pull request merged**